# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1281203272 14400
#      Sat Aug 07 13:47:52 2010 -0400
# Node ID d19d2abdd65f1fde10f9489a0258ad2160ce6805
# Parent  5a55773fcaa93f19dfa6f1c53efda3204a840b96
__fltk_print__.m: When running pstoedit, crop eps file after it is created.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-08-07  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__fltk_print__.m: When running pstoedit, crop eps file after it
+	is created. Correctly check error status.
+
 2010-08-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/cell2mat.m: Allow cells of struct arrays.
 
 2010-08-05  Ben Abbott <bpabbott@mac.com>
 
 	* /plot/__print_parse_opts__.m, plot/__fltk_print__.m: Don't warn on
 	absence of fig2dev or pstoedit until they are needed.
diff --git a/scripts/plot/__fltk_print__.m b/scripts/plot/__fltk_print__.m
--- a/scripts/plot/__fltk_print__.m
+++ b/scripts/plot/__fltk_print__.m
@@ -144,55 +144,55 @@ function status = __fig2dev__ (opts, fig
   persistent warn_on_absence = true
   if (nargin < 4)
     devfile = opts.name;
   endif
   if (nargin < 3)
     devopt =  opts.devopt;
   endif
   if (! isempty (opts.fig2dev_binary))
-    cmd = sprintf ("%s -L %s %s %s 2>&1", opts.fig2dev_binary, devopt, figfile, devfile);
+    cmd = sprintf ("%s -L %s %s %s 2>&1", opts.fig2dev_binary, devopt,
+                   figfile, devfile);
     [status, output] = system (cmd);
-    if (opts.debug)
+    if (opts.debug || status != 0)
       fprintf ("fig2dev command: %s", cmd)
     endif
-    if (status)
+    if (status != 0)
+      disp (output)
       warning ("print:fig2devfailed", "print.m: error running fig2dev.")
-      disp (cmd)
-      disp (output)
     endif
   elseif (isempty (opts.fig2dev_binary) && warn_on_absence)
     warning ("print:nofig2dev", "print.m: 'fig2dev' not found in EXEC_PATH.")
     warn_on_absence = false;
   endif
 endfunction
 
 function status = __pstoedit__ (opts, devopt, name)
   persistent warn_on_absence = true
   if (nargin < 3)
     name = opts.name;
   endif
   if (nargin < 2)
     devopt =  opts.devopt;
   endif
+  if (! isempty (opts.pstoedit_binary))
     tmp_epsfile = strcat (tmpnam (), ".eps");
+    drawnow ("eps", tmp_epsfile)
     if (opts.tight_flag)
       __tight_eps_bbox__ (opts, tmp_epsfile);
     endif
-  if (! isempty (opts.pstoedit_binary))
-    drawnow ("eps", tmp_epsfile)
-    cmd = sprintf ("%s -f %s %s %s 2>&1", opts.pstoedit_binary, devopt, tmp_epsfile, name);
+    cmd = sprintf ("%s -f %s %s %s 2>&1", opts.pstoedit_binary, devopt,
+                   tmp_epsfile, name);
     [status, output] = system (cmd);
-    if (opts.debug)
+    if (opts.debug || status != 0)
       fprintf ("pstoedit command: %s", cmd)
     endif
-    if (status)
+    if (status != 0)
+      disp (output)
       warning ("print:pstoeditfailed", "print.m: error running pstoedit.")
-      disp (cmd)
-      disp (output)
     endif
     [status, output] = unlink (tmp_epsfile);
     if (status != 0)
       disp (output)
       warning ("print.m: failed to delete temporay file, '%s'.", tmp_epsfile)
     endif
   elseif (isempty (opts.pstoedit_binary) && warn_on_absence)
     warning ("print:nopstoedit", "print.m: 'pstoedit' not found in EXEC_PATH.")
